### DISCUSSION TITLE
issue: Multiple Renders Occurred

### DISCUSSION THREAD

**Discussion by rajagopal1997** [react-hook-form/react-hook-form#9347] [Q&A] [answered]
### Version Number

7

### Codesandbox/Expo snack

https://codesandbox.io/s/reacthookformmultiplerendering-03hgyc

### Steps to reproduce

1. Go to [Codesandbox](https://codesandbox.io/s/reacthookformmultiplerendering-03hgyc)
2. Enter some text then remove that text, that time multiple renders are occurred
3. how to control unwanted renders in
4. Check renders counts in header (I'll Send sample video)
5. if getting unwanted renders then performance will be reduced

https://user-images.githubusercontent.com/83446783/200819620-4da466d5-d5ea-465c-b035-e09cc0e01177.mp4

### Expected behaviour

Control unwanted renders, increase performance

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
please check if this is caused by React strict mode.

**Reply by rajagopal1997** [reply]
i am not using react strict mode ...please check my sandbox

**Reply by rajagopal1997** [reply]
please check video

**Comment by rajagopal1997** [comment]
how to solve that issue, please give me some solution

**Comment by leapful** [comment, marked as answer]
@rajagopal1997 

There is nothing wrong on the example. You are using hook `useForm` inside root component App, therefore input field is valid or invalid, `formState.errors` object from react-hook-form is changed too (add or remove error message) and triggers re-rendering.

**Comment by rajagopal1997** [comment]
@leapful Thank you for giving your replay...
